### ISSUE TITLE
Loader component does not have accessible name (JAWS)

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#423]
NVDA and Voiceover both announce the `label` prop set for `aria-valuetext`, but JAWS completely skips the component not announcing its state. With an actual [progress bar](https://cauldron.dequelabs.com/components/ProgressBar), it feels like this component should be more generic.

**Comment by scurker** [comment]
We should make the following changes:

* Remove `role="progressbar"` and associated props
* Only set to `role="alert"` when `aria-label` or `aria-labelledby` is set
* Set `aria-hidden` when there's no label
* `aria-label` or `aria-labelledby` are optional
